rpcs3/Emu/Cell/SPURecompiler.cpp

```diff
@@ -7355,7 +7355,7 @@ class spu_llvm_recompiler : public spu_recompiler_base, public cpu_translator
 			{
 				const u32 exponent = data._u32[i] & 0x7f800000u;
 
-				if (data._u32[i] > 0x7f7fffffu || !exponent)
+				if (data._u32[i] >= 0x7f7fffffu || !exponent)
```

Comment:
Nothing was really wrong in this line, in fact it can be even written as `if (data._u32[i] > 0x7f800000u || !exponent)`.

Reply:
In theory there's nothing wrong with `if (data._s32[i] < 0 || !exponent)` but we produce "extended range" values differently than real hardware so it might not always be safe. If we had 100% accurate softfloat for all other instructions it'd be fine, but it's safer to match the output that the "non constant" path will produce I think.

Likewise, `>= 0x7f7fffff` vs `> 0x7f7fffff` is similar, and this one doesn't really matter, but for my piece of mind I'd rather match the "non constant" path behavior here too.

Reply:
Actually I just remembered, I originally wrote it in a similar way (Checked for any positive number that isn't 0), but there were issues in a lot of games. It's specifically because it was producing different results for values larger than `0x7f7fffff`, changing it to the current implementation fixed all of those issues.

Reply:
Added a new comment so it doesn't get forgotten in the future.